### DISCUSSION TITLE
supabase access to storage object content via SQL

### DISCUSSION THREAD

**Discussion by davidaventimiglia-professional** [supabase/supabase#11357] [Questions] [answered]
Hi!

I see that supabase storage can be accessed in SQL via the storage schema and its buckets and objects tables, but is there any way to access the content of a storage object via SQL? What I mean is, is there any way to insert or read the contents of a storage object via a SQL query?

**Comment by GaryAustin1** [comment, marked as answer]
You would need to use something like the pgsql-http or pg_net extensions to do a request to the storage REST API.

**Reply by davidaventimiglia-professional** [reply]
Cool thanks!

**Reply by GaryAustin1** [reply]
An example using http extension of what I use to delete a storage file by passing a function the pathname and bucket.
```
declare
  myurl varchar := 'https://xxxxxxxxxxxxxxxxxxxxxxxxx.supabase.co';
  service_role_key varchar := 'longservicerolekey';
  url varchar := myurl || '/storage/v1/object/' || bucket || '/' || pathname;
begin
return (
select status FROM http((
          'DELETE',
            url,
           ARRAY[http_header('authorization', 'Bearer ' || service_role_key)],
           NULL,
           NULL
        )::http_request)
);
end;
```
